File: ctci/chapter_1_arrays_strings/js/1.3-URLify.js
/* O(n) solution */

/**
 * Replaces all spaces with "%20".
 * @param   {string}  str
 * @returns {string}
 */
const urlify = str => {
  return str.trim().replace(/ /g, "%20")
}

console.log(urlify("Mr John Smith    "))
/* Mr%20John%20Smith */
console.log(urlify("Say hello to my little friend          "))
/* Say%20hello%20to%20my%20little%20friend */

File: ctci/chapter_1_arrays_strings/1.3-URLify.py
# O(n) solution


def urlify(string: str) -> str:
    """
    Replaces all spaces with "%20".
    """
    return string.strip().replace(' ', '%20')


if __name__ == '__main__':
    print(urlify("Mr John Smith    "))
    # Mr%20John%20Smith
    print(urlify("Say hello to my little friend          "))
    # Say%20hello%20to%20my%20little%20friend
